### ISSUE TITLE
(How) can I pass objects to protected eval?

### ISSUE THREAD

**Issue by wolfgangr** [yaroslaff/evalidate#10]
I want to implement a controlled Python eval to implement user definable functions for FreeCAD spreadsheets.
( see here https://github.com/FreeCAD/FreeCAD/issues/14042#issuecomment-2143963786)

The arguments to my eval are not bare strings, but may be python objects (numbers, Lists, Positions, even fullfledged CAD objects)
For this purpuse, here... 
https://stackoverflow.com/questions/21100203/passing-arguments-to-python-eval
I found this hack (maybe it's a feature of Python, but haven't ever seen it before)

```rv = eval(funcnam)(*params)```

Where `params` is a **mixed list of arbitrary python objects.**

Can I do this with evalidate as well?

**Comment by wolfgangr** [comment]
ah, further debugging, I found the trick:
The arguments are not even subject to eval in this syntax.
eval(name) expands to the 'real' function name, and only then (and not subject to eval) the argument list is passed.

I seperated the syntax into three distinct steps (just to inspect what happens), and then it works:
```
    func_evd = evalidate.Expr(funcnam, model=sPyMod_model)
    func_ptr = func_evd.eval()
    rv = func_ptr(*params)
```
The first two statements still could even stay combined.
I still don't understand in detail what happens. But I might assume that "eval" builtin gets special handling in the parser, while evalidate's pendants do not?
